.github/workflows/third-party.yml

```diff
@@ -596,6 +596,51 @@ jobs:
 
       - name: Run Dify unit tests
         run: uv run --project api bash dev/pytest/pytest_unit_tests.sh
+  test-cadwyn:
+    name: Test Cadwyn (main branch) on Python ${{ matrix.python-version }}
+    # If 'schedule' was the trigger, don't run it on contributors' forks
+    if: |
+      github.event_name != 'schedule' ||
+      (github.event_name == 'schedule' && github.repository == 'pydantic/pydantic')
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
+        # Only test on 3.12 if the PR wasn't labeled as 'third-party-tests':
+        exclude:
+          - python-version: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'third-party-tests') && '3.9' }}
+          - python-version: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'third-party-tests') && '3.10' }}
+          - python-version: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'third-party-tests') && '3.11' }}
+          - python-version: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'third-party-tests') && '3.13' }}
+    steps:
+    - name: Checkout Cadwyn
+      uses: actions/checkout@v4
+      with:
+        repository: zmievsa/cadwyn
+
+    - name: Checkout Pydantic
+      uses: actions/checkout@v4
+      with:
+        path: pydantic-latest
+
+    - uses: astral-sh/setup-uv@v6
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Cadwyn dependencies
+      run: |
+        uv venv --python ${{ matrix.python-version }}
```

Comment:
```suggestion
```

Can we use https://github.com/astral-sh/setup-uv?tab=readme-ov-file#activate-environment instead?

Reply:
Done!